GetIntoTeachingApi/Models/BaseModel.cs

```diff
@@ -260,8 +260,23 @@ private void MapRelationshipAttributesToEntity(Entity source, ICrmService crm, O
                 var attribute = EntityRelationshipAttribute(property);
                 var value = property.GetValue(this);
 
-                if (attribute == null || value == null)
+                if (attribute == null)
+                {
+                    continue;
+                }
+
+                var relationship = new Relationship(attribute.Name);
+
+                if (value == null)
                 {
+                    context.LoadProperty(source, relationship);
+
+                    if (source.RelatedEntities.ContainsKey(relationship))
+                    {

```

Comment:
This would only work when there can only be one related entity (event -> building)